src/community/leading_eigenvector.c

```diff
@@ -1005,31 +1006,33 @@ int igraph_le_community_to_membership(const igraph_matrix_t *merges,
     igraph_vector_t fake_memb;
     long int components, i;
 
-    if (igraph_matrix_nrow(merges) < steps) {
```

Comment:
`igraph_le_community_to_membership()` is part of the public API (although I don't quite understand why, but this ship has sailed), so in theory anyone can call this function with weird merge matrices. Therefore, this check should be kept, even though it is already checked in the "normal" route where we call `igraph_le_community_to_membership()` from within the library.

Reply:
Every call to le_community_to_membership will also call community_to_membership. This call is done before anything else is done with `steps` or `merges`. So community_to_membership will always do the check, and le_community_to_membership will fail properly without this test. The same thing would simply be tested twice with these lines added.

Reply:
Okay, fair enough, I misunderstood this part then!